12-watchtower-API.md

```diff
@@ -22,227 +22,320 @@ The scope of this bolt does not include:
+* MUST set `qos_len` equal to the length of `qos_data`.
 
-`modes` define the operation mode requested / accepted. Modes include `altruistic` and `non-altuistic`.
+The receiving node:
 
-`qos` defines whether the sender is requesting / accepting Quality of Service for his appointments. The only QoS offered at the moment is `accountability`.
+The receiving node MUST reject the appointment if:
+* Authentication is required and `auth_token` is not provided.
+* Authentication is required and `auth_token` is invalid.
+* `locator` is not a `16-byte` value.
+* `start_block` is further than one block behind the current chain tip.
+* `start_block` is further than one block ahead the current chain tip.
+* `encrypted_blob` has unreasonable size.
+* `cipher` is not among the ones he implements.
 
-#### Requirements
-The sending node: 
-
-* MUST send `wt_init` as the first message.
-* MUST set `client_server` to the corresponding value, depending on whether it is a client or a server.
-* MUST set `accepted_ciphers` to the list of ciphers he implements.
-* MUST set `modes` to the list of modes that he is willing to accept.
-* SHOULD set `qos` to the the quality of service he is requesting / offering. 
-
-The requesting node: 
-
-* MUST receive `wt_init` before sending any other message.
-* MUST respond with its own `wt_init` message
-* upon receiving a `client_server` field that matches its own (a server receives a `wt_init` from a server or a client receives one from a client):
-	* MUST fail the connection
-* upon receiving an `accepted_ciphers` that does not contain any of its own accepted ciphers:
-	* MUST fail the connection
-* upon receiving a `modes` that does not contain any of its own accepted modes:
-	* MUST fail the connection
-* upon receiving a `qos` that does not contain any of its own accepted QoS:
-	* MUST fail the connection
+The receiving node SHOULD reject the appointment if:
+
+* `end_block` is too far away in the future.
+
+The receiving node MUST: 
+
+* truncate the remainder of the package to `qos_len`.
+* if `qos_len` is not 0:
+	* process `qos_data` according to [Quality of Service data](#quality-of-service-data).
+
+The receiving node MAY accept the appointment otherwise.
 
 #### Rationale
 
-The client is always the one in charge of establishing the connection. His `wt_init` message informs the server of what type of service he is requesting. The server should agree with the client if the implement the same methods. Otherwise the connection should be failed.
+WatchTowers may work in different modes (e.g. altruistic vs non-altruistic). In some of those modes, proof of payment may be required for the WatchTower to perform provide the service. `auth_token` is used to decide wether the user is entitled to use the service or not in the cases it may be required. Notice that the tokens do not need to be linked to any kind of identity but confirm that the payment has been performed.
```

Comment:
typo `perform provide`.

Reply:
I would also change:
`Notice that the tokens do not need to be linked to any kind of identity but confirm that the payment has been performed.`
to
`Notice that the tokens do not need to be linked to any kind of identity, but act as proof that the payment has been performed.`